data-workflows/activity/install_activity_model.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import time
+from datetime import datetime, timezone
+from enum import Enum
+from typing import Callable, List, Union
+import os
+
+from pynamodb.models import Model
+from pynamodb.attributes import UnicodeAttribute, NumberAttribute
+
+from utils.utils import get_current_timestamp
+
+LOGGER = logging.getLogger()
+
+
+def to_utc_timestamp_in_millis(timestamp: datetime) -> int:
+    return int(timestamp.replace(tzinfo=timezone.utc).timestamp() * 1000)
+
+
+class InstallActivityType(Enum):
+
+    def __new__(cls, value, timestamp_formatter, type_timestamp_formatter):
+        install_activity_type = object.__new__(cls)
+        install_activity_type._value = value
+        install_activity_type.timestamp_formatter = timestamp_formatter
+        install_activity_type.type_timestamp_formatter = type_timestamp_formatter
+        return install_activity_type
+
+    DAY = (1, to_utc_timestamp_in_millis, lambda timestamp: f'DAY:{timestamp.strftime("%Y%m%d")}')
+    MONTH = (2, to_utc_timestamp_in_millis, lambda timestamp: f'MONTH:{timestamp.strftime("%Y%m")}')
+    TOTAL = (3, lambda timestamp: None, lambda timestamp: 'TOTAL:')
```

Comment:
I'm still not a fan of overloading the enum with methods and complex values like this.

How about keeping this as a simple enum
```suggestion
    DAY = auto()
    MONTH = auto()
    TOTAL = auto()
```

then making separate dicts to map the types to the formats

also instead of using lambdas, I'd suggest using the built-in string formatting:
```python
type_timestamp_formats = {
   InstallActivityType.DAY: 'DAY:{0:"%Y%m%d"}',
   InstallActivityType.MONTH: 'MONTH:{0:"%Y%m"}',
   InstallActivityType.MONTH.TOTAL: 'TOTAL:',
}

type_format = type_timestamp_formats[my_activity_type].format(my_timestamp)
```

Reply:
I feel keeping that a part of the enum closely couples the enum to this formatting. As this formatting is specific to each type of the enum. I think making it a map, adds distance between that relationship. What do you think?